### DISCUSSION TITLE
Consume environment variables properly within nuxt.config.js

### DISCUSSION THREAD

**Discussion by gplusdotgr** [nuxt/nuxt#9533] [Questions (Nuxt 2)]
We're trying to consume environment variables to configure Sentry, and we find that it makes a difference (the variables are either undefined or have the correct value) depending on where we add the configuration.

When the configuration is on nuxt.config.js, [as proposed in the docs](https://sentry.nuxtjs.org/guide/setup/) the env vars are undefined. 
```js
 sentry: {
   dsn: process.env.SENTRY_AUTH_TOKEN
   config: {
     environment: process.env.STAGE
   }
 }
```

When we move the configuration in a separate external file, (a module), things work.

When trying to debug, we were wondering also how to properly log or step-debug to see what the values are during nuxt building..

**Comment by agnelnieves** [comment]
Hey! Are you guys defining the env variables using the `env` key in nuxt.config?

```
{
...
  env: {
    vimeoClientId: process.env.VIMEO_CLIENT_ID,
  },
 ... 
}
```

**Reply by agnelnieves** [reply]
Actually, just realized you meant in the config for sentry within `nuxt.config`. You should be able to consume the env variables there too. Make sure it has the correct env file wherever you're deploying
